File: 백준/Bronze/26676. Wybór zadań/Wybór zadań.c
#include <stdio.h>
int main() {
    int n, c[15] = {0}, i, r, d;
    scanf("%d", &n);
    char p[3];
    while (n--) {
        scanf("%s", p);
        r = (p[0] - '1') * 3;
        d = p[1] - 'A';
        c[r + d]++;
    }
    for (i = 0; i < 15; i++)
        if (c[i] < 1 + (i / 3 == 4))
            return !printf("NIE\n");
    printf("TAK\n");
}

File: 백준/Bronze/26676. Wybór zadań/Wybór zadań.py
def c(p):
    A = {f"{B}{C}": 0 for B in range(1, 6) for C in 'ABC'}
    for E in p:
        if E in A: A[E] += 1
    for B in range(1, 5):
        for C in 'ABC':
            if A[f"{B}{C}"] < 1: return "NIE"
    return "NIE" if any(A[f"5{C}"] < 2 for C in 'ABC') else "TAK"
input()
print(c(input().split()))
